mirascope/core/base/_utils/_setup_call.py

```diff
@@ -16,21 +16,24 @@
 _BaseToolT = TypeVar("_BaseToolT", bound=BaseTool)
 _BaseDynamicConfigT = TypeVar("_BaseDynamicConfigT", bound=BaseDynamicConfig)
 
+# We can't set TypedDict as a bound for a TypeVar, so we use Mapping instead
+_BaseCallParamsT = TypeVar("_BaseCallParamsT", bound=Mapping)
+
 
 def setup_call(
     fn: Callable[..., _BaseDynamicConfigT | Awaitable[_BaseDynamicConfigT]],
     fn_args: dict[str, Any],
     dynamic_config: _BaseDynamicConfigT,
     tools: list[type[BaseTool] | Callable] | None,
     tool_type: type[_BaseToolT],
-    call_params: BaseCallParams,
+    call_params: _BaseCallParamsT,
 ) -> tuple[
-    str,
+    str | None,
     list[BaseMessageParam | Any],
     list[type[_BaseToolT]] | None,
-    dict[str, Any],
+    _BaseCallParamsT,
```

Comment:
maybe what we want here is actually for `BaseCallKwargs` to take a generic for the call params that we unpack.

The idea here is the `call_kwargs` is a superset of `call_params` and can generally be expected to include all of the items from `call_params` in `call_kwargs` (except in instances where we remove certain keys due to certain settings such as removing tool choice or updating it).

Reply:
Fixed